src/wrappers/tomopy/recon.py

```diff
@@ -58,13 +59,15 @@ def rotation(params: Dict, method_name:str, comm: Comm, data: ndarray) -> float:
     float
         The center of rotation.
     """
+   
     module = getattr(recon, 'rotation')
     method_func = getattr(module, method_name)
     rot_center = 0
     mid_rank = int(round(comm.size / 2) + 0.1)
     if comm.rank == mid_rank:
-        mid_slice = data.shape[1] // 2
-        rot_center = method_func(data[:, mid_slice, :], **params)
+        if params['ind'] == 'mid':
```

Comment:
Actually we don't need a preview as such here as we've got **ind** parameter from TomoPy which specifies the index. I just enabled here also 'mid' to be the middle index of the dataset.